packages/pigeon/lib/dart_generator.dart

```diff
@@ -107,8 +108,28 @@ String _makeGenericCastCall(TypeDeclaration type, String nullTag) {
       : '';
 }
 
-String _getArgumentName(NamedType field) =>
-    field.name.isEmpty ? 'arg' : field.name;
+String _getArgumentName(int count, NamedType field) =>
+    field.name.isEmpty ? 'arg$count' : field.name;
+
+String _spaceJoin(String x, String y) => '$x $y';
+
+Iterable<String> _getMethodArgumentNames(Method func) =>
+    indexMap(func.arguments, _getArgumentName);
+
+String _getMethodArgumentsSignature(
+  Method func,
+  Iterable<String> argNames,
+  String nullTag,
+) {
+  return func.arguments.isEmpty
+      ? ''
+      : map2(
+          func.arguments
+              .map((NamedType e) => _addGenericTypes(e.type, nullTag)),
+          argNames,
+          _spaceJoin,
+        ).join(', ');
```

Comment:
While writing the above, I realized that I'm confused about this method at a higher level: I was wondering why the argument names were a separate iterator in the first place, and pieced together the various calls, and unless I'm mis-reading you are:
- Starting with `func.arguments`
- Having the caller pull out just the names via _getMethodArgumentNames to create `argNames`
- Having this pull out just the types via line 127
- Zipping them back into one piece at the end.

Why isn't this a function that just takes `func.arguments` and `nullTag` as parameters, and constructing the string directly from func.arguments in a simple iterator? Doing the two halves separately, and at different layers of the code, seems like it's just making it unnecessarily hard to follow what's happening.

Reply:
> I was wondering why the argument names were a separate iterator in the first place, and pieced together the various calls

If I'm understanding the question correctly, the reason that `argNames` is a parameter is that it guarantees that the same argument names are used here that are used elsewhere in the code.  See line 164:
```dart
        final Iterable<String> argNames = _getMethodArgumentNames(func);
        sendArgument = '<Object>[${argNames.join(', ')}]';
        argSignature = _getMethodArgumentsSignature(func, argNames, nullTag);
```

`argNames` is used twice for 2 different calculations.  Does that answer your question?  Technically `_getMethodArgumentsSignature` could call `_getMethodArgumentNames` but that would be a duplication of that logic that could create problems in the future.

Reply:
We should discuss this offline tomorrow.

(To preview at a high level: it is not clear to me what future problem that could create that is worse than allowing someone to call `_getMethodArgumentsSignature` where `_argNames` is anything other than exactly `_getMethodArgumentNames(func)`, which your current structure makes trivial. E.g.,, nothing would stop someone from calling `_getMethodArgumentsSignature(someFunc, _getMethodArgumentNames(aCompletelyDifferentFuncWithTheSameArgCount), nullTag)` with incredibly bad results.)

Reply:
Done.  I talked with stuart offline.  His biggest concern was relying on the 2 input data structures being passed into `_getMethodArgumentsSignature` having to mirror each other.  My concern was having different places in the code that aren't next to each other that have to match.  By making `_getMethodArgumentsSignature` a higher order function we could solve both problems.  Now it looks like this:
```dart
        final Iterable<String> argNames =
            indexMap(func.arguments, _getArgumentName);
        sendArgument = '<Object>[${argNames.join(', ')}]';
        argSignature =
            _getMethodArgumentsSignature(func, _getArgumentName, nullTag);
```
If I wanted to be extra secure I could pull _getArgumentName into a local variable and share it, but this is fine.